### PR TITLE
refactored grouped stops

### LINKED ISSUE #35: Refactor grouped stops YAML structure
right now, we define a grouped stop like
```yaml
grouped_stops:
  - operator: CT
    group_name: San Jose Diridon Caltrain Station
    ids:
      - 70261
      - 70262
```

the above implies that the operator for a grouped stop must be the same for all ids. but what if we were dealing with a stop that had multiple operators (like VTA and Caltrain)?

lets refactor the grouped stops to be like below, where the below is identical to the above value
```yaml
grouped_stops:
  - name: San Jose Diridon Caltrain Station
    stops:
    - operator: CT
      id: 70261
    - operator: CT
      id: 70262
```

### game plan
- [ ] try to parse the above yaml values with the library we use https://pyyaml.org/wiki/PyYAMLDocumentation
- [ ] refactor the server to accept the new format for grouped stops, test locally with your own data.yaml

later we can refactor how we define fixed stops, as well as move the hardcoded stop id to coordinates mapping to its own file

### PR DESCRIPTION
**Pull request by cuzethan**
resolves #35

### CODE REVIEW

**Comment by steeevin88** [comment]
could you update the README too, thanks

File: server/app.py [hunk 1]
```diff
@@ -94,14 +94,15 @@ def update_cache():
         new_stops.append(stop_info)
 
     for group in grouped_stops:
-        stop_info = get_stop_predictions(group.get('ids'), group.get('operator'), group.get('group_name'), group.get('use_destination_as_name', False))
+        for stop in group.get("stops"):
```
Comment (evanugarte) [inline server/app.py]: we can do this, that way if stops is null, we wont get the below error
```suggestion
        for stop in group.get("stops", []):
```

```
Traceback (most recent call last):
  File "/server/app.py", line 216, in helper_thread
    update_cache()
  File "/server/app.py", line 97, in update_cache
    for stop in group.get("stops"):
TypeError: 'NoneType' object is not iterable
```

**Comment by cuzethan** [comment]
good to merge?

File: server/app.py [hunk 2]
```diff
@@ -94,14 +94,15 @@ def update_cache():
...
-            for timetable in group.get('timetables', []):
-                if predictions := get_fixed_stops_predictions(now, timetable.get('route_name'), timetable.get('destination')):
-                    stop_info.predictions.append(predictions)
+            if group.get('timetables'):
+                for timetable in group.get('timetables', []):
+                    if predictions := get_fixed_stops_predictions(now, timetable.get('route_name'), timetable.get('destination')):
+                        stop_info.predictions.append(predictions)
 
-        new_stops.append(stop_info)
+            new_stops.append(stop_info)
```
Comment (evanugarte) [inline server/app.py]: i dont think we need the if statement, we can just do the `.get` with a default value of an empty list, meaning we wont loop at all if timetables is None

like
```py
for timetable in group.get('timetables', []):
    if predictions := get_fixed_stops_predictions(now, timetable.get('route_name'), timetable.get('destination')):
        stop_info.predictions.append(predictions)

new_stops.append(stop_info)

```

File: README.md [hunk 3]
```diff
@@ -25,14 +28,17 @@ sce link transit
...
-      group_name: San Jose Diridon Caltrain Station
-      ids:
-        - 70261
-        - 70262
+    - name: San Jose Diridon Caltrain Station
+      stops:
+        - operator: CT
+          id: 70261
+        - operator: CT
+          id: 70262
```
Comment (evanugarte) [inline README.md:36]: can we undo the newlines that were added to this file, the only changes should be this section

File: server/app.py [hunk 4]
```diff
@@ -94,14 +94,14 @@ def update_cache():
         new_stops.append(stop_info)
 
     for group in grouped_stops:
-        stop_info = get_stop_predictions(group.get('ids'), group.get('operator'), group.get('group_name'), group.get('use_destination_as_name', False))
+        for stop in group.get("stops", []):
+            stop_info = get_stop_predictions([stop.get('id')], stop.get('operator'), group.get('name'), group.get('use_destination_as_name', False))
```
Comment (evanugarte) [inline server/app.py]: i know i am nit picking here but lets make the args on their own lines, easier to read
```suggestion
            stop_info = get_stop_predictions(
                [stop.get('id')],
                stop.get('operator'),
                group.get('name'),
                group.get('use_destination_as_name', False),
            )
```

File: server/app.py [hunk 5]
```diff
@@ -94,14 +94,14 @@ def update_cache():
         new_stops.append(stop_info)
 
     for group in grouped_stops:
-        stop_info = get_stop_predictions(group.get('ids'), group.get('operator'), group.get('group_name'), group.get('use_destination_as_name', False))
+        for stop in group.get("stops", []):
+            stop_info = get_stop_predictions([stop.get('id')], stop.get('operator'), group.get('name'), group.get('use_destination_as_name', False))        
 
-        if group.get('timetables'):
             for timetable in group.get('timetables', []):
```
Comment (evanugarte) [inline server/app.py]: since we removed the if statement, dont we have to unindent this line?

**Comment by cuzethan** [comment]
finished final changes

**Pull request merged**